frontend/packages/console-shared/src/hooks/useRoutesWatcher.ts

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import { K8sResourceKind, RouteKind } from '@console/internal/module/k8s';
+import { useK8sWatchResources } from '@console/internal/components/utils/k8s-watch-hook';
+import { getRoutesForServices, getServicesForResource } from '../utils';
+
+export const useRoutesWatcher = (
+  resource: K8sResourceKind,
+): { loaded: boolean; loadError: string; routes: RouteKind[] } => {
+  const { namespace } = resource.metadata;
+  const [loaded, setLoaded] = React.useState<boolean>(false);
+  const [loadError, setLoadError] = React.useState<string>('');
+  const [routes, setRoutes] = React.useState<RouteKind[]>();
+  const watchedResources = React.useMemo(
+    () => ({
+      services: {
+        isList: true,
+        kind: 'Service',
+        namespace,
+      },
+      routes: {
+        isList: true,
+        kind: 'Route',
+        namespace,
+      },
+    }),
+    [namespace],
+  );
```

Comment:
Could not find a usage of `useRoutesWatcher` which also uses the fetched services. Can we remove the services resource lookup here?
And can we use then `useK8sWatchResource` instead of `useK8sWatchResources` then?

Reply:
The call to `getServicesForResource` requires the services to be loaded, the returned services are used to determine the routes to be returned.

Reply:
Didn't see that, thanks for the explanation :heavy_check_mark: